python/src/rt_m1_apps/msaf_configuration.py

```diff
@@ -475,6 +517,21 @@ async def sync_configuration(m1: M1Session, streams: dict) -> dict:
             if crc is not None:
                 if not await m1.consumptionReportingConfigurationCreate(ps_id, crc):
                     log_error("Failed to activate ConsumptionReportingConfiguration for Provisioning Session %s")
+            
+            if metrics is not None:
+                if isinstance(metrics,dict):
+                    metrics_list = metrics.items()
+                elif isinstance(metrics,list):
+                    metrics_list = [(m.get('samplingPeriod', None), m) for m in metrics]
+                else:
+                    log_error(f'Configured metrics for provisioning session "{cfg_id}" should be an object or array')
+                    metrics_list = None
+                if metrics_list is not None:
+                    for sampling_period, metricsconfig in metrics_list:
+                        result = await m1.metricsReportingConfigurationCreate(ps_id, metricsconfig)
```

Comment:
The style here seems to have been copied from the dynamic policies. In dynamic policies it made sense to allow the configuration to specify the mapping of unique external reference to the policy details, but I'm not sure it makes sense to specify metrics using a unique sampling period here.

So, in dynamic policies, the following would be equivalent configurations:
```json
"policies": {
    "basic": {
        "qoSSpecification": {
            "maxAuthBtrUl": "20 Kbps",
            "maxAuthBtrDl": "2 Mbps"
        }
    },
    "high-quality": {
        "qoSSpecification": {
            "maxAuthBtrUl": "40 Kbps",
            "maxAuthBtrDl": "10 Mbps"
        }
    }
}
```
... and ...
```json
"policies": [
    {
        "externalReference": "basic",
        "qoSSpecification": {
            "maxAuthBtrUl": "20 Kbps",
            "maxAuthBtrDl": "2 Mbps"
        }
    },
    {
        "externalReference": "high-quality",
        "qoSSpecification": {
            "maxAuthBtrUl": "40 Kbps",
            "maxAuthBtrDl": "10 Mbps"
        }
    }
]
```

The former just helping enforce the uniqueness of the external reference (as properties of an object in json have to use unique keys).

But this implementation of the the metrics configuration would allow:
```json
"metricsReporting" : {
    "1": {
        "scheme":"urn:scheme1",
        "reportingInterval":5,
        "samplePercentage":33
    },
    "2": {
        "scheme": "urn:scheme2",
        "reportingInterval":10,
        "samplePercentage":66
    }
}
```
...which isn't very useful unless each metrics scheme you want to specify uses a different sampling period. You wouldn't be able to use this to describe two different schemes which have the same sampling period.

Therefore I think you should simplify this and drop "metricsReporting" as a `dict`, and just use a `list` of metrics reporting objects.